kernel/src/udp_port_table.rs

```diff
@@ -0,0 +1,249 @@
+//! UDP port table implementation for enforcing port binding. This table is
+//! checked when packets are sent/received by a UDPSendStruct/UDPReceiver.
+//! UdpPortBinding provides an opaque descriptor object that allows the holder
+//! interact with the bound port table. Only the holder of the UdpPortBinding
+//! object can interact with its own corresponding location in the bound port
+//! table. In order to bind to a particular port as sending/receiving, one must
+//! obtain the corresponding sender/receiving binding from UdpPortBinding.
+use crate::returncode::ReturnCode;
+use core::fmt;
+use tock_cells::optional_cell::OptionalCell;
+use tock_cells::take_cell::TakeCell;
+
+//#![allow(dead_code)]
+const MAX_NUM_BOUND_PORTS: usize = 16;
+
+/// The PortEntry struct is stored in the table and conveys what port is bound
+/// at the given index if one is bound. If no port is bound, the value stored
+/// at location is Unbound.
+#[derive(Clone, Copy, PartialEq)]
+pub enum PortEntry {
+    Port(u16),
+    Unbound,
+}
+
+// We need Option<PortEntry> to distinguish between the case in which we have
+// a UdpPortSocket that is not bound to a port and an index where there is no
+// UdpPortSocket allocated.
+static mut PORT_TABLE: [Option<PortEntry>; MAX_NUM_BOUND_PORTS] = [None; MAX_NUM_BOUND_PORTS];
```

Comment:
I'm not sure I understand this data structure. What do the entries correspond to?